ExtensionStore/lib/store.js

```diff
@@ -1036,14 +1036,15 @@ LocalExtensionList.prototype.checkFiles = function (extension) {
 
 /**
  * Installs the extension
+ * @param {QToolButton} widget - Optional widget to use as a progressbar.
  * @returns {bool}  the success of the installation
  */
-LocalExtensionList.prototype.install = function (extension) {
+LocalExtensionList.prototype.install = function (extension, widget) {
```

Comment:
I'm a bit worried about this, it kinds of intertwines the UI, the backend, and the styling... I feel like there must be a better way to separate all these concerns, but I'm not sure how just yet, since we can't create custom signals.

for example, I feel like maybe we could set up the extensionDownloader to do the link/reporting?